source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -874,6 +874,20 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       request_headers_->setPath(new_path);
     }
 
+    if (request_headers_->size() > request_headers_->maxHeadersCount()) {
+      ENVOY_STREAM_LOG(
+          debug,
+          "ext_authz filter rejected the request because the header count limit would be exceeded",
+          *decoder_callbacks_);
+      decoder_callbacks_->streamInfo().setResponseFlag(
+          StreamInfo::CoreResponseFlag::UnauthorizedExternalService);
```

Comment:
`UnauthorizedExternalService` might not be the best but i will leave this to @ggreenway